packages/discord.js/src/structures/ApplicationRoleConnectionMetadata.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+class ApplicationRoleConnectionMetadata {
+  constructor(data) {
+    /**
+     * The name of this metadata field
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The name localizations for this metadata field
+     * @type {?Object<Locale, string>}
+     */
+    if ('name_localizations' in data) {
+      this.nameLocalizations = data.name_localizations;
+    } else {
+      this.nameLocalizations ??= null;
+    }
```

Comment:
This pattern is only applicable for `_patch`, in the constructor it should be:

```suggestion
this.nameLocalizations = data.name_localizations ?? null;
```

Reply:
Same applies below.